glean-core/src/database/mod.rs

```diff
@@ -443,6 +534,14 @@ impl Database {
             .write()
             .expect("Can't access app lifetime data as writable")
             .clear();
+
+        let ping_lifetime_data = &self.ping_lifetime_data;
```

Comment:
When we do `for lifetime in [Lifetime::User, Lifetime::Ping].iter()` on line 522, if the deferred write is on, we don't need to also clear the RKV store. Maybe not a big deal, but could save us some I/O.

Reply:
This is going to require some refoactoring of this function and I don't know if it makes sense to do it, because if I do that it will be removed in my next PR.